File: findNumbersDisappeared.java
/** Time Complexity: O(n) where n is the number of elements in array
 * Space Complexity: O(1)
 * Executed on Leetcode: yes
 * approach :
 * we iterate through the array, we check if the number present at the index = Math.abs(nums[i]-1) is marked negative or not, if not,we mark that number negative
 * we iterate the array once again and return the indices of the elements which are still positive.
 */
import java.util.*;
class findNumbers
{
    public List<Integer> findmissing(int[] nums)
    {   List<Integer> list = new LinkedList<>();
        if(nums.length==0) return list;
        for(int i=0;i<nums.length;++i)
        {
            int checkIndex = Math.abs(nums[i])-1;
            if(nums[checkIndex]>0) nums[checkIndex] *= -1;
        }
        for(int i=0;i<nums.length;++i)
        {
            if(nums[i]>0) list.add(i+1);
        }
        return list;
    }
    public static void main(String[] args) {
        findNumbers obj = new findNumbers();
        int[] nums = new int[]{4,3,2,7,8,2,3,1};
        System.out.println("missing numbers:"+obj.findmissing(nums));
    }
}

File: Find_numbers_disappeared.py

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        #Defining the output list
        list= []

        #Looping over the input array 
        # to mark all values greater than 0 as negative
        for i in nums:
            pos = abs(i) -1
            if nums[pos] > 0:
                nums[pos] *= -1

        #Looping over the length of the array to append values into output array
        for i in range(len(nums)):
            if nums[i] > 0:
                list.append(i+1)
        return list
        
#This problem finds the missing elements in the given array corresponding to the length of the array

#Time complexity - O(N)
#Space complexity - O(1)

nums = [4,3,2,7,8,2,3,1]
obj = Solution()
print(obj.findDisappearedNumbers(nums))